recipes/configs/llama2/1B_full_ppo.yaml

```diff
@@ -0,0 +1,182 @@
+  ]
+  # this is the only place where you should update `recipe_checkpoint` if resuming training
+  recipe_checkpoint: null
+  output_dir: ${output_dir}/policy
+  model_type: LLAMA2
+
+# this should be setup identically to the policy model checkpointer at the start of training
+# ensure `checkpoint_files` always points to the original policy weights, even if resuming training
+ref_policy_checkpointer:
+  _component_: torchtune.utils.FullModelHFCheckpointer
+  checkpoint_dir: /tmp/TinyLlama_v1.1
+  checkpoint_files: [
+      "pytorch_model.bin",
+  ]
+  output_dir: ${output_dir}/policy
+  model_type: LLAMA2
+
+# checkpointer for the value model - update this if resuming from checkpoint
+# since this model will be identical to the reward model it's helpful to initialise this
+# from the trained reward model weights
+value_checkpointer:
+  _component_: torchtune.utils.FullModelHFCheckpointer
+  checkpoint_dir:  /tmp/tinyllama_rm_sentiment_1b
+  # only `checkpoint_files` need to be updated if resuming training
+  checkpoint_files: [
+      "model.safetensors"
+  ]
+  output_dir: ${output_dir}/value
+  model_type: MISTRAL_REWARD
+
+# checkpointer for the reward model, ensure `checkpoint_files`
+# always points to the original reward model weights, even if resuming training
+reward_checkpointer:
+  _component_: torchtune.utils.FullModelHFCheckpointer
+  checkpoint_dir:  /tmp/tinyllama_rm_sentiment_1b
+  checkpoint_files: [
+      "model.safetensors"
+  ]
+  output_dir: ${output_dir}/value
+  model_type: MISTRAL_REWARD
+
+
+# Training env
+device: cuda
+
+# Training arguments
+batch_size: 256
+num_steps: 100000
+ppo_epochs: 2
+ppo_batch_size: 128
+gradient_accumulation_steps: 8
+
+# Memory management and performance
+compile: True
+optimizer:
+  _component_: torch.optim.AdamW
+  weight_decay: 0.01
+  lr: 3e-6
+
+optimizer_in_bwd: False
+log_peak_memory_stats: False
+enable_activation_checkpointing: True
+
+# Reduced precision
+dtype: bf16
+
+# Trajectory generation arguments
+
+# batch size for forward pass during generation
+forward_batch_size: 32
+max_generated_tokens: 58
+temperature: 0.7
+top_k: null
+
+# Reward args
+
+# parameter for penalising generations shorter than `min_response_length`
+min_response_length: 18
+# parameter for penalising generations without a stop token
+penalise_no_eos: True
+# scalar penalty to apply when penalising
+reward_penalty: -3
+
+# tokens to consider as "end of sequence" tokens
+stop_token_ids: [
+  2,  # eos_id
+  29889 # llama2 "." token
+]
+whiten_rewards: False
+# GAE hyperparameters
+gamma: 1
+lmbda: 0.95
+
+# PPO hyperparameters
+loss:
+  _component_: torchtune.modules.loss.PPOLoss
+  epsilon: 0.2
+  value_coeff: 0.1
+  value_clip_range: 0.2
+kl_coeff: 0.01
+
+
+# Logging
+metric_logger:
+  _component_: torchtune.utils.metric_logging.DiskLogger
+  log_dir: ${output_dir}
+
+log_every_n_steps: 1
+
+resume_from_checkpoint: False
+output_dir: /tmp/llama2-1b-ppo-finetune
+seed: null
+shuffle: True
```

Comment:
In general, I think we want to keep configs as simple as possible and leave any complexity not necessarily needed to code or maybe just by showing an example in a tutorial. 182 lines for a config feels uncomfortable. Are there any other simplifications we can make here? Orr is there anything torchtune can do to help make the config simpler for a use case like this?

cc @kartikayk  and @ebsmothers